### ISSUE TITLE
hs-boot is not handled correctly

### ISSUE THREAD

**Issue by lspitzner** [ndmitchell/ghcid#78]
An error in Foo.hs-boot it is not reported by `ghcid`, instead i get "All good".

Tested in manual `cabal repl`, i see the error even after reloading, so i think `ghc` is doing its job correctly.

I have no minimal example but could set something up if desired.

**Comment by lspitzner** [comment]
Everything works correctly in the presence of hs-boot stuff iff the hs-boot module is clean of errors.

**Comment by ndmitchell** [comment]
I have to map modules and filenames to each other, I suspect this is just a case of matching on `.hs` somewhere that `.hs-boot` is also required. My next task for ghcid is to get a more complete test suite up and running, with more varied examples of launching commands, and the error in `.hs-boot` seems a very suitable test case. I'll try adding it there are fixing it.

**Comment by ndmitchell** [comment]
If you load a working setup, then break `hs-boot`, then hit `:reload` without changing anything else, do you see any errors?

**Comment by lspitzner** [comment]
(using `cabal repl` for testing)

indeed, i can confirm that no error is produced. Even after `:quit; cabal repl; :reload` (!)

**Comment by lspitzner** [comment]
Only thing to fix it is `clean`. and i think a `cabal build` does it too, for some reason.

This is ghc-8.0.1 btw.

(but i was not completely mistaken above, was i? could you reproduce some repl-works, ghcid-does-not case?)

**Comment by ndmitchell** [comment]
I have managed to reproduce once or twice, and reach a few corner cases I haven't managed to get back to. It's clear there are quite a few ghci bugs in this area which isn't helping. I suspect I'll be able to improve things though.

**Comment by ndmitchell** [comment]
I've reproduced things, but nothing consistent, and nothing that works the same for more than about 2 minutes. Often changing the boot file causes nothing to rebuild at all. And then all of a sudden changing it reliably causes `:reload` to work. I can't figure it out. If you are regularly changing the boot file, you are going to end up with problems in `ghci`, and `ghcid` will inherit many of them.

That said, I did find one instance of misparsing a warning, and I was ignoring messages from boot files even when they were produced, so it should be better now. Not good, but hopefully no worse than the underlying `ghci`.
